### ISSUE TITLE
How use ytpl with proxy ?

### ISSUE THREAD

**Issue by Murena7** [TimeForANinja/node-ytpl#102]
Hi all - is possible use ytpl with proxy ? Please show example how i can do this ? Thanks

**Comment by TimeForANinja** [comment]
it should work the same way it works for ytdl-core
extract from the official ytdl-core example:
```js
const ytdl = require('..');
const HttpsProxyAgent = require('https-proxy-agent');

// Remove 'user:pass@' if you don't need to authenticate to your proxy.
const proxy = 'http://user:pass@111.111.111.111:8080';
const agent = HttpsProxyAgent(proxy);

const stream = ytdl('https://www.youtube.com/watch?v=aqz-KE-bpKQ', {
  requestOptions: { agent },
});
```

and for ytpl:
```js
const ytpl = require('ytpl');
const HttpsProxyAgent = require('https-proxy-agent');

// Remove 'user:pass@' if you don't need to authenticate to your proxy.
const proxy = 'http://user:pass@111.111.111.111:8080';
const agent = HttpsProxyAgent(proxy);

const playlist = await ytpl('<some id in here>', {
  requestOptions: { agent },
});
```

**Comment by Murena7** [comment]
I try pass agent - same as in example but this didn't work - i receive error
TypeError [ERR_INVALID_ARG_TYPE] [ERR_INVALID_ARG_TYPE]: The "options.agent" property must be one of Agent-like Object

But when i do same in ytdl-core - all work fine, may be in ytdl-core library some different  ?

**Comment by TimeForANinja** [comment]
should be fixed with https://github.com/TimeForANinja/node-ytpl/commit/30763ff962492aa2cefd971dbe89eef6a6419190 and the new v2.0.4 release

**Comment by Murena7** [comment]
> should be fixed with [30763ff](https://github.com/TimeForANinja/node-ytpl/commit/30763ff962492aa2cefd971dbe89eef6a6419190) and the new v2.0.4 release

This great news - thank you so much !
